### PR TITLE
update contributing guide

### PR DESCRIPTION
**Pull request by mengqiy**

### CODE REVIEW

**Comment by mengqiy** [comment]
cc: @bharathkkb

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -77,14 +77,65 @@ For each function, its files spread in the follow places:
...
+supported options [here][e2e test harness doc].
 
 You can choose to put the e2e test in either the `examples/` directory or in the
 `tests/` directory depending on if it is worthwhile to be shown as an example.
 
+To test all examples and e2e tests, run
+
+```shell
+$ make e2e-test
+```
```
Comment (bharathkkb) [inline CONTRIBUTING.md:111]: I think most folks may not need this. IIRC this took +15m

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -77,14 +77,65 @@ For each function, its files spread in the follow places:
...
+#### Doc Verifier
+
+We have a script to ensure the usage docs and the examples are consistent.
+Please ensure it's passing by running:
+
+```shell
+$ ./scripts/verify-docs.py
+```
+
+This script requires Python 3 and `mdrip` which is a CLI tool. To
```
Comment (bharathkkb) [inline CONTRIBUTING.md:129]: Also had to pip install pyyaml

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -77,14 +77,65 @@ For each function, its files spread in the follow places:
...
+```shell
+$ make unit-test
+```
+
+#### E2E Tests
 
 The e2e tests are the recommended way to test functions in the catalog. They are
-very easy to write and set up with our [e2e test harness].
+very easy to write and set up with our e2e test harness. You can find all the
+supported options [here][e2e test harness doc].
```
Comment (bharathkkb) [inline CONTRIBUTING.md:102]:
```suggestion
supported options and expected test directory structure [here][e2e test harness doc].
```

**Review by bharathkkb — APPROVED** [review]
Thanks for adding this! Minor comments but LGTM

File: CONTRIBUTING.md (hunk 1 above)
Comment (mengqiy) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 2 above)
Comment (mengqiy) [reply CONTRIBUTING.md]: Thanks for pointing this out!

**Comment by mengqiy** [comment]
@bharathkkb Thanks for your feedback! PTAL

**Review by droot — APPROVED** [review]

**Pull request merged**